celery/platforms.py

```diff
@@ -153,7 +153,7 @@ def acquire(self):
 
     def is_locked(self):
         """Return true if the pid lock exists."""
-        return os.path.exists(self.path)
+        return os.path.exists(self.path) and not self.remove_if_stale()
```

Comment:
I don't see this method used anywhere else, so you can redefine its purpose. However, `is_*` seem more informative and here there are side-effects. How about renaming it also to `maybe_unlock`?

Reply:
Although unlikely that users depend on this class, that is a very good point, thanks for taking the time to consider. I suggested the renaming for future readers and failed to take into account this possibility.

You could always revert and just mark `is_locked` as deprecated, by adding a deprecation warning (see `celery.utils.deprecated.warn` function) and simply create a new function (protected or public) with the name of your choice and the new functionality. What do you think?

Reply:
Just noticed your latest commit, seems like a good solution. Let me have one final look please.